plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/contributors/FirCallableCompletionContributor.kt

```diff
@@ -167,11 +184,79 @@ internal open class FirCallableCompletionContributor(
         val typeOfPossibleReceiver = explicitReceiver.getKtType() ?: return
         val possibleReceiverScope = typeOfPossibleReceiver.getTypeScope() ?: return
 
+        if (explicitReceiver is KtSuperExpression) {
```

Comment:
Please, create additional position context for super references and move all logic related to that into separate completion contributor to make the code more readable